pkg/router/functionHandler.go

```diff
@@ -614,6 +614,7 @@ func (fh functionHandler) getServiceEntryFromCache() (serviceUrl *url.URL, err e
 			}
 			errMsg = fmt.Sprintf("Error getting function %v;s service entry from cache: %v", fh.function.ObjectMeta.Name, err)
 		}
+		fh.logger.Info("404 getting function", zap.String("error_message", errMsg))
```

Comment:
This isn't a 404, right underneath it is returning `http.StatusInternalServerError` which is 500.